src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java

```diff
@@ -161,18 +163,19 @@ public abstract Object getInstance(Class<?> pluggableClass)
      * @return the default plugins used for this instance of this descriptor as a list of Object. Used for
      * help/doc generation.
      */
-    public abstract List<Object> getDefaultInstances();
+    public abstract List<? super T> getDefaultInstances();
 
     /**
      * @return an ordered List of actual plugin instances that have been specified on the command
      * line, in the same order they were obtained/created by {@line #getInstance}).
      */
-    public abstract List<T> getAllInstances();
+    public abstract List<? super T> getAllInstances();
 
     /**
      * @param pluginName Name of the plugin requested
      * @return Class object for the plugin instance requested
      */
+    // TODO: this method is unused in Barclay and GATK -> it is something needed?
     public abstract Class<?> getClassForInstance(final String pluginName);
```

Comment:
I used it in an interim version of docgen, I believe because I needed access to the full classname of the plugIn in order to generate an html link. So, I think its another case where it should remain as is.